hazelcast/src/main/java/com/hazelcast/nio/ClassLoaderUtil.java

```diff
@@ -192,6 +192,27 @@ public static boolean isClassDefined(String className) {
         }
     }
 
+    /**
+     * Check whether given class implements an interface with the same name.
+     * It returns true even when the implemented interface is loaded by a different
+     * classloader and hence the class is not assignable into it.
+     *
+     * This is useful for logging purposes.
+     *
+     * @param clazz
+     * @param iface
+     * @return
+     */
+    public static boolean implementsIntefaceWithSameName(Class<?> clazz, Class<?> iface) {
+        Class<?>[] interfaces = clazz.getInterfaces();
```

Comment:
Afaik, even visiting superclasses is not enough. You should also obtain interfaces extended by its own interfaces and superclasses' interfaces.